### PR TITLE
warn if BigQuery failed rows collection is not processed

### PR DESCRIPTION
**Pull request by reuvenlax**

### CODE REVIEW

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowRunnerTest.java [hunk 1]
```diff
@@ -2384,7 +2389,93 @@ public void visitPrimitiveTransform(@UnknownKeyFor @NonNull @Initialized Node no
...
+
+  public void testBigQueryDLQWarning(BigQueryIO.Write.Method method, boolean processFailures)
+      throws IOException {
+    PipelineOptions options = buildPipelineOptions();
+    List<String> experiments =
+        new ArrayList<>(
+            ImmutableList.of(
+                GcpOptions.STREAMING_ENGINE_EXPERIMENT,
+                GcpOptions.WINDMILL_SERVICE_EXPERIMENT,
+                "use_runner_v2"));
```
Comment (bvolpato) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowRunnerTest.java:2429]: nit: experiments irrelevant here?

**Review by bvolpato — APPROVED** [review]
LGTM.

This `traverseTopologically` / `enterCompositeTransform` is cool, thanks Reuven!

**Review by bvolpato — APPROVED** [review]
LGTM, just `./gradlew spotlessApply` first

**Comment by bvolpato** [comment]
Actually it's not only apply, need to rename the generic type

```
23:20:19 [ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Spotless_Commit/src/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StreamingWriteTables.java:306:11: Name 'TupleTagType' must match pattern '^(((T|K|V|W|X)[0-9]*)|([A-Z][a-z][a-zA-Z]*T))$'. [MethodTypeParameterName]
```

**Comment by bvolpato** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by liferoad** [comment]
R: @Abacn

**Comment by liferoad** [comment]
R: @shunping-google

**Pull request merged**